pennylane_lightning/src/tests/lightning_gates_unittest.cpp

```diff
@@ -0,0 +1,159 @@
+// Copyright 2021 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "gtest/gtest.h"
+#include "../rework/Gates.hpp"
+#include "GateData.h"
+#include "TestingUtils.h"
+
+#include <functional>
+
+using std::unique_ptr;
+using std::vector;
+using std::string;
+using std::function;
+
+using Pennylane::CplxType;
+
+namespace test_gates{
+
+const vector<double> ZERO_PARAM = {};
+const vector<double> ONE_PARAM = {0.123};
+const vector<double> THREE_PARAMS = {0.123, 2.345, 1.4321};
```

Comment:
(See comment below). Should we add additional values here?

Reply:
We could, although personally, I wouldn't worry too much about it. These tests simply check that the matrices are okay, we could test applying the gates more extensively. Having said that I could add more params if others also agreed that it's important.

Reply:
Fair